lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -631,6 +632,71 @@ def remove_obsolete_script_phases(removed_phase_names = REMOVED_SCRIPT_PHASE_NAM
           end
         end
 
+        # Updates all user targets to include on demand resources specified by libraries. Note that currently,
+        # only app level targets are allowed to include on demand resources.
+        #
+        # @return [void]
+        #
+        def add_on_demand_resources
```

Comment:
this method looks a bit complicated but its not. Its trying relatively to avoid adding duplicate file references.

OnDemandResources of a target are added to the users target because they need to be part of the resources_build_phase and also included in the "Resource Tags".

Note as of right now ODRs can only be used by app targets so if a pod specifies ODR paths then those are integrated into the target the pod is integrated with.